Add Formal provider to community package list

thatliuser opened the pull request:
- [x] Released your package with a version prefixed with `v` (e.g. `v0.1.0`). The
      part after the leading `v` must be valid semver 2.0.

    - Published an SDK for your provider in:
        - [x] Typescript
        - [x] Python
        - [x] Go
        - [x] C#
        - [ ] Java (optional)

- [x] Have checked in a schema.json that matches the location of the `schemaFile`
      specified in `/community-packages/package-list.json`.

- [x] Have a `/docs/_index.md` and `/docs/installation-configuration.md` filled
      out in your repo.

    - [x] `/docs/installation-configuration.md` links to all published SDKs.

    - [x] `/docs/index.md` shows an example of using your provider in each language.

Once you are ready to publish, opening a PR will tag a member of Pulumi who will
review your PR.

Thanks for contributing to the Pulumi Registry!

This adds the Formal provider for Pulumi. The only item in the checklist that isn't satisfied is a published C# SDK since we don't have a customer need for this at the moment. Please let me know if I need to change anything for this to be approved, thank you!

dschaller commented:
To keep quality in the Pulumi Registry high, we have a check-list before merging a new provider into the registry.

- [ ] Pulumi has appropriate contact information from the provider maintainer

  If the provider is community maintained (maintained by a person, not a company), then a GitHub handle is sufficient

  If the provider is maintained by a company, Pulumi needs a contact person at the maintaining company.

- [ ] The package will generate accurate documentation:

  1. Check out the PR under review and run:

  ```sh
  $ make bin/resourcedocsgen
  $ ./bin/resourcedocsgen metadata from-github \
          --repoSlug '<repoSlug>' \
          --schemaFile '<schemaFile>' \
          --version '<version>'
  ```

  Here `<repoSlug>` and `<schemaFile>` should match exactly the values added to `/community-packages/package-list.json`.

  This will generate metadata for the provider locally.

  1. Push the metadata files into a PR (either back to the PR under review or a new PR).

  After pushing the provider to CI and waiting for a preview site:

  - [ ] Confirm that that CI passes **for the PR with the metadata files**.

  - [ ] Click through the site preview and confirm that the docs (for the new provider) render as expected.

  - [ ] The registry renders a valid logo for the new provider.

- [ ] Hand-written docs are complete and accurate:

  - [ ] Validate that you can install the new provider with the instructions found in `/docs/installation-configuration.md`.

  Maintainers should run the `pulumi plugin install resource <name> <version> --server <pluginDownloadURL>` command specified in the `/docs/installation-configuration.md` and see a provider be downloaded.

  - [ ] `/docs/installation-configuration.md` contains a link to the published SDK in each language (i.e. TypeScript, Python, Go and C#).

  - [ ] `/docs/_index.md` contains a minimal example in every supported language.
  - [ ] `/docs/_index.md` contains a brief explanation of what the package does.

- [ ] There is a published version:
  - [ ] The repository has a version tag prefixed with `v` that corresponds with a valid GitHub release
  - [ ] Each published SDK has a matching release

- [ ] A CODEOWNER has approved the PR.

dschaller commented:
Hi @thatliuser, thanks for the contribution! I've added our checklist for community contributions in a comment above and will be going through the criteria there so you can track.

For our community plugins we request that your SDK also support C#. Can you generate an SDK for that language as well?

thatliuser commented:
Hi @dschaller, I've updated the provider to also publish itself on NuGet and added the appropriate docs as requested in the checklist. For a point of contact, you can use my email (eric at joinformal dot com); I emailed Jack today and gave him some additional details there. Is there anything else I need to do before this can be reviewed?

dschaller commented:
Thanks! Let me give your updated branch a pull and work through the rest of the checklist

dschaller commented:
@thatliuser thanks! I've approved the changes but I'm still waiting to hear back on the team about how to handle merging this PR since CI can't properly run against forks of the repo (it can't access the secrets we rely on to perform things like a site preview). I'll let you know when I hear back.

thatliuser commented:
Hi @dschaller, are there any updates on this?

dschaller commented:
Merged in https://github.com/pulumi/registry/pull/8369
